Week2/nadin.js

```diff
@@ -0,0 +1,181 @@
+
+//5
+3 === 3 ? console.log ('yes') : console.log('no');
+
+//6
+
+function vehicle(color, code, age) {
+    if (age === 5 && code === 1){
+        console.log('a' + ' ' + color + ' ' + 'used car');
+        return;
+    } else if (age === 5 && code === 2){
+        console.log('a' + ' ' + color + ' ' + 'used motorbike');
+        return;
+    }
+    switch (code) {
+        case 1: 
+        console.log('a' + ' ' + color + ' ' + 'car');
+        break;
+        case 2: 
+        console.log('a' + ' ' + color + ' ' + 'motorbike');
+        break;
+        default:
+        console.log('parameter not supported');
+    }
+ }
+
+ vehicle("blue", 1, 5);
+ //a blue used car
+
+ vehicle("blue", 2, 5);
+
+//7 
+const vehicles = ["motorbike", "caravan", "bike"];
+
+//8
+console.log(vehicles[2]);
+
+//9
+function vehicle(color, code, age) {
+    if (age === 1){
+        console.log('a' + ' ' + color + ' ' + 'new' + ' ' + vehicles[code-1]);
+    } else {
+        console.log('a' + ' ' + color + ' ' + 'used' + ' ' + vehicles[code-1]);
+    } 
+ }
+ //10
+const vehicles = ["motorbike", "caravan", "bike"];
+let advertisement = "Amazing Joe's Garage, we service cars, "
+for (let i = 0; i < vehicles.length; i++) {
+     if(i === vehicles.length - 1) {
+         advertisement += "and " + vehicles[i] + "s";
+     } else {
+        advertisement += vehicles[i] + "s, ";
+     }
+ }
+console.log(advertisement);
+
+//11
+const vehicles = ["motorbike", "caravan", "bike"];
+let advertisement = "Amazing Joe's Garage, we service cars, "
+vehicles.push("bus");
+for (let i = 0; i < vehicles.length; i++) {
+    if(vehicles[i].slice(-1) === "s"){
+        vehicles[i] += 'e';
+    }
+    if(i === vehicles.length - 1) {
+        advertisement += "and " + vehicles[i] + "s";
+    } else {
+       advertisement += vehicles[i] + "s, ";
+    }
+}
+console.log(advertisement);
+
+
+//12
+const obj = {};
+//13 14
+const teachersName = {
+        teachers: ['philip', 'Unmesh', 'Yash'],
+        language1: ['Html','Css','Git','Javascript'],
+        }
+    
+//15
+let x = [1, 2, 3];
+let y = [1, 2, 3];
+let z = y;
+    console.log('x == y', x == y, "they do not have the same reference");
+    console.log('x === y', x === y, "they do not have the same reference");
+    console.log("z == y", z == y, "they have the same reference" );
+    console.log('z == x', z == x, "they do not have the same reference");
+
+//16
+let o1 = { foo: "bar" };
+let o2 = { foo: "bar" };
+let o3 = o2;
+console.log({o1}, {o2}, {o3});
+o2 = { foo2: "bar2" };
+console.log({o2}, {o3});
+o1 =  { foo1: "bar1" };
+console.log( {o1}, {o3});
+ 
+o3 = o2; 
+o2 = o3;
+console.log("It dose matter because o3 is undefined and 02 is already declared");
+
+//17
+let bar = 42;
+typeof typeof bar;
+
+console.log(typeof typeof bar);
+console.log("typeof typeof bar: is string because we asked about the type of the word 'bar' which is string not about the type of the value of the variable bar which is number");
```

Comment:
👍

Reply:
Thanks for your feedback! It is done! 👍 :)